.github/workflows/python-package.yml

```diff
@@ -39,6 +40,24 @@ jobs:
         poetry env info
         poetry run pip list
 
+    # Style/format checks.
+    - name: Run Black (code formatter)
+      run: |
+        poetry run black --check .
+
+    - name: Run isort (import sorting)
+      run: |
+        poetry run isort --check-only --profile black .
+
+    # TODO: "pyflakes[F]" failed during execution due to RecursionError('maximum recursion depth exceeded')
+    # - name: Run Flake8 (style guide enforcement)
+    #   run: |
+    #     poetry run flake8 .
+
+    - name: Run Ruff (linter)
+      run: |
+        poetry run ruff check .
+
     - name: Test with pytest
       run: |
         # run tests in tests/ dir and only fail if there are failures or errors
```

Comment:
Moving the cache step before installing dependencies might help pip restore with the cached packages, avoiding redownloading dependencies in following workflow runs. Test this thought I'm not 100%.